### PR TITLE
Add a --cleanup option

### PR DESCRIPTION
**Pull request by dennisschagt**
Resolves https://github.com/newsboat/newsboat/issues/1182.

### CODE REVIEW

File: doc/chapter-firststeps.asciidoc [hunk 1]
```diff
@@ -21,6 +21,7 @@ usage: ./newsboat [-i <file>|-e] [-u <urlfile>] [-c <cachefile>] [-x <command> .
     -E, --export-to-file=<file>     export list of read articles to <file>
     -I, --import-from-file=<file>   import list of read articles from <file>
     -h, --help                      this help
+        --cleanup                   remove unreferenced items from cache
```
Comment (Minoru) [inline doc/chapter-firststeps.asciidoc:24]: Please change the Newsboat version in this listing to 2.22 (which is the next scheduled release). Otherwise someone might read it and think this was available since 2.10.

File: doc/configcommands.dsv [hunk 2]
```diff
@@ -17,7 +17,7 @@ confirm-exit||[yes/no]||no||If set to `yes`, then Newsboat will ask for confirma
 cookie-cache||<path>||""||Set a cookie cache. If set, cookies will be cached in (i.e. read from and written to) this file, using http://www.cookiecentral.com/faq/#3.5[Netscape format].||cookie-cache "~/.newsboat/cookies.txt"
 datetime-format||<date/time format>||%b %d||This format specifies the date/time format in the article list. For a detailed documentation on the allowed formats, consult the manpage of strftime(3).||datetime-format "%D, %R"
 define-filter||<name> <filterexpr>||n/a||With this command, you can predefine filters, which you can later select from a list, and which are then applied after selection. This is especially useful for filters that you need often and you don't want to enter them every time you need them.||define-filter "all feeds with 'fun' tag" "tags # \"fun\""
-delete-read-articles-on-quit||[yes/no]||no||If set to `yes`, then all read articles will be deleted when you quit Newsboat.||delete-read-articles-on-quit yes
+delete-read-articles-on-quit||[yes/no]||no||If set to `yes`, all read articles will be deleted when you start Newsboat with the `--cleanup` option or quit Newsboat if `cleanup-on-quit` is set.||delete-read-articles-on-quit yes
```
Comment (Minoru) [inline doc/configcommands.dsv:20]: That's a very complicated sentence, it's hard to parse correctly even if one is familiar with the code :) I'd split it in two, something like this:

> If set to `yes`, all read articles will be deleted when you run Newsboat with `--cleanup` option, or when you quit it normally. This setting only applies when `cleanup-on-quit` is set.

This is still quite complicated, so I'd be glad if you can improve it further.

File: doc/manpage-newsboat.asciidoc [hunk 3]
```diff
@@ -38,6 +38,12 @@ terminals on Unix or Unix-like systems such as GNU/Linux, BSD or Mac OS X.
         _delete-read-articles-on-quit_, _keep-articles-days_, and _max-items_
         settings.
 
+--cleanup::
+        Remove unreferenced entries from the cache. Feeds and their articles
+        will be removed if the feedurl is no longer in the _urls_ file. If
```
Comment (Minoru) [inline doc/manpage-newsboat.asciidoc:43]: When I first read this, I got confused a bit: why would we delete *read* articles if we already deleted *all* of them? I think the reason for the confusion is that there's two separate thoughts in this (single) paragraph.

My advice is to splitt the last sentence into a paragraph of its own, and make it say "*all* read articles will also be deleted", to further emphasize that *this* sentence applies to *all* feeds, not just the ones that are gone from the urls file.

**Review by Minoru — CHANGES_REQUESTED** [review]
The code looks great! I have a few comments about the docs though. I'm not sure how to best address them, so please propose your alternatives if you can think of any.

And thanks for working on this! I'm happy to see that you're keeping the momentum ^_^

File: doc/configcommands.dsv (hunk 2 above)
Comment (dennisschagt) [reply doc/configcommands.dsv:20]: > This setting only applies when cleanup-on-quit is set.

That sounds incorrect because the setting also applies when only specifying `--cleanup` (even when `cleanup-on-quit` would be `no`).

Do you think the following is clear enough?
> If set to `yes`, all read articles will be deleted when quiting Newsboat. This option only applies if `cleanup-on-quit` is set to `yes` or if the `--cleanup` argument is passed.

File: doc/manpage-newsboat.asciidoc (hunk 3 above)
Comment (dennisschagt) [reply doc/manpage-newsboat.asciidoc]: Good idea, done: https://github.com/newsboat/newsboat/pull/1250/commits/f03bcc83298e98b2c802b6e8770ca2aefa13f53f

File: doc/chapter-firststeps.asciidoc (hunk 1 above)
Comment (dennisschagt) [reply doc/chapter-firststeps.asciidoc:24]: Done: https://github.com/newsboat/newsboat/pull/1250/commits/7f30aa737b28ba48974d2f3a0e2c5b64e8e809b8

File: doc/configcommands.dsv (hunk 2 above)
Comment (Minoru) [reply doc/configcommands.dsv:20]: > That sounds incorrect because …

Ah, yes, you're right! Your new suggestion does sound much clearer, let's go with it.

**Review by Minoru — APPROVED** [review]
Excellent, thank you very much @dennisschagt!
